src/java/org/apache/cassandra/index/sai/disk/v1/LongArray.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.v1;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import org.apache.cassandra.index.sai.SSTableQueryContext;
+import org.apache.cassandra.index.sai.disk.v1.bitpack.BlockPackedReader;
+
+/**
+ * Abstraction over a long-indexed array of longs.
+ */
+public interface LongArray extends Closeable
+{
+    /**
+     * Get value at {@code idx}.
+     */
+    long get(long idx);
+
+    /**
+     * Get array length.
+     */
+    long length();
+
+    /**
+     * Using the given token returns the first row ID corresponding to the token.
+     * @param targetToken Token to lookup and it must not be smaller than previous value
+     * @return The row ID of the given token or negative value if target token is greater than all tokens
+     */
+    long findTokenRowID(long targetToken);
```

Comment:
nit: If this is meant to be a general utility, what if we just call it `long indexOf(long value)` or something that fits with other array-based list stuff in Java?